### PR TITLE
feat(dialog): add afterOpen to MdDialogRef

### PR DESCRIPTION
**Pull request by willshowell**
Provides a hook for initializing component after animation is
complete.

Relates to https://github.com/angular/material2/issues/3616#issuecomment-327398901.

### CODE REVIEW

File: src/lib/dialog/dialog-ref.ts [hunk 1]
```diff
@@ -44,6 +46,13 @@ export class MdDialogRef<T> {
     private _containerInstance: MdDialogContainer,
     public readonly id: string = `md-dialog-${uniqueId++}`) {
 
+    // Emit when opening animation completes
+    RxChain.from(_containerInstance._animationStateChanged)
+      .call(filter, event => event.phaseName === 'done' && event.toState === 'enter')
+      .call(first)
+      .subscribe(() => this._afterOpen.next());
```
Comment (jelbourn) [inline src/lib/dialog/dialog-ref.ts]: Also `complete`?

Comment (willshowell) [reply src/lib/dialog/dialog-ref.ts]: Ah yep, fixed

File: src/lib/dialog/dialog-ref.ts [hunk 2]
```diff
@@ -30,11 +29,14 @@ export class MdDialogRef<T> {
   /** Whether the user is allowed to close the dialog. */
   disableClose = this._containerInstance._config.disableClose;
 
+  /** Subject for notifying the user that the dialog has finished opening. */
+  private _afterOpen = new Subject<void>();
+
   /** Subject for notifying the user that the dialog has finished closing. */
-  private _afterClosed: Subject<any> = new Subject();
+  private _afterClosed = new Subject<any>();
```
Comment (crisbeto) [inline src/lib/dialog/dialog-ref.ts]: These subjects should be `Subject<void>`.

Comment (willshowell) [reply src/lib/dialog/dialog-ref.ts]: Don't they carry the close value of the dialog?

**Pull request merged**